In [1]:
# Import libraries
import pandas as pd
import os
from dotenv import load_dotenv

# Load the ipython-sql extension
%load_ext sql

In [2]:
# Load .env file with PostgreSQL server credentials
dotenv_path = os.path.join(os.getcwd(), 'DB_URL.env')
load_dotenv(dotenv_path)

# Configure the connection using the DB_URL environment variable
%sql $DB_URL

# DEVELOPED AND DEVELOPING COUNTRIES

In [3]:
%%sql

-- Total countries by status
SELECT status,
	   COUNT(DISTINCT country) AS country_tally,
	   ROUND(CAST(COUNT(DISTINCT country) AS NUMERIC) / (SELECT CAST(COUNT(DISTINCT country) AS NUMERIC)
								 					       FROM world_life_expectancy
                                                         ) * 100, 1) AS percent_total
  FROM world_life_expectancy
 GROUP BY status
;

 * postgresql://postgres:***@localhost:5433/world_life_expectancy
2 rows affected.


[('Developed', 32, Decimal('17.5')), ('Developing', 151, Decimal('82.5'))]

This dataset encompasses a total of 183 countries. Approximately 83% of these nations fall under the category of developing countries, while the remaining 17% are classified as developed countries.

In [4]:
%%sql

-- Proportion of developed and developing countries by region
SELECT region,
	   COUNT(DISTINCT CASE WHEN status = 'Developed' THEN country END) AS developed_country_tally,
	   ROUND(COUNT(DISTINCT CASE WHEN status = 'Developed' THEN country END) / (SELECT CAST(COUNT(DISTINCT country) AS NUMERIC)
								 					       						  FROM world_life_expectancy
																				 WHERE status = 'Developed'
                                                         						) * 100, 1) AS developed_country_percent_total,
	   COUNT(DISTINCT CASE WHEN status = 'Developing' THEN country END) AS developing_country_tally,
	   ROUND(COUNT(DISTINCT CASE WHEN status = 'Developing' THEN country END) / (SELECT CAST(COUNT(DISTINCT country) AS NUMERIC)
								 					       						   FROM world_life_expectancy
																				  WHERE status = 'Developing'
                                                         						 ) * 100, 1) AS developing_country_percent_total
  FROM world_life_expectancy
 GROUP BY region
 ORDER BY region
;

 * postgresql://postgres:***@localhost:5433/world_life_expectancy
5 rows affected.


[('Africa', 0, Decimal('0.0'), 54, Decimal('35.8')),
 ('Americas', 1, Decimal('3.1'), 32, Decimal('21.2')),
 ('Asia', 2, Decimal('6.3'), 45, Decimal('29.8')),
 ('Europe', 27, Decimal('84.4'), 12, Decimal('7.9')),
 ('Oceania', 2, Decimal('6.3'), 8, Decimal('5.3'))]

- Europe is region with the largest share of developed countries, making up approximately 84% of the total.

- On the other hand,  Africa is the region with the highest percentage of developing countries, accounting for about 36% of the total.

In [5]:
%%sql

-- Life expectancy summary statistics by status
SELECT status,
	   COUNT(DISTINCT country) AS country_tally,
	   MIN(life_expectancy) AS min_life_expectancy,
	   MAX(life_expectancy) AS max_life_expectancy,
	   PERCENTILE_CONT(0.5) WITHIN GROUP (ORDER BY life_expectancy) AS median_life_expectancy,
	   ROUND(CAST(AVG(life_expectancy) AS NUMERIC), 1) AS avg_life_expectancy,
	   ROUND(CAST(STDDEV(life_expectancy) AS NUMERIC), 1) AS std_life_expectancy
  FROM world_life_expectancy
 GROUP BY status
;

 * postgresql://postgres:***@localhost:5433/world_life_expectancy
2 rows affected.


[('Developed', 32, 69.9, 89.0, 79.25, Decimal('79.2'), Decimal('3.9')),
 ('Developing', 151, 36.3, 89.0, 69.0, Decimal('67.1'), Decimal('9.0'))]

- Over time, developed countries consistently exhibit a notably higher minimum life expectancy, averaging around 70 years, compared to developing countries, where the minimum hovers around 36 years.
  
- Both developed and developing countries share a common maximum life expectancy of 89 years, indicating that, despite differences in minimum values, there is a shared upper limit in life expectancy across nations.

- In terms of average life expectancy, developed countries surpass their developing counterparts, boasting an average of approximately 79 years. In contrast, developing countries, despite constituting a larger proportion of nations, have an average life expectancy of around 67 years.

In [6]:
%%sql

-- Mortality summary statistics by status
SELECT status,
	   COUNT(DISTINCT country) AS country_tally,
	   MIN(adult_mortality) AS min_adult_mortality,
	   MAX(adult_mortality) AS max_adult_mortality,
	   PERCENTILE_CONT(0.5) WITHIN GROUP (ORDER BY adult_mortality) AS median_adult_mortality,
	   ROUND(CAST(AVG(adult_mortality) AS NUMERIC), 1) AS avg_adult_mortality,
	   ROUND(CAST(STDDEV(adult_mortality) AS NUMERIC), 1) AS std_adult_mortality
  FROM world_life_expectancy
 GROUP BY status
;

 * postgresql://postgres:***@localhost:5433/world_life_expectancy
2 rows affected.


[('Developed', 32, 1, 229, 73.0, Decimal('79.7'), Decimal('47.9')),
 ('Developing', 151, 1, 723, 163.0, Decimal('182.8'), Decimal('128.0'))]

- Both developed and developing countries exhibit an identical minimum adult mortality rate of 1, showcasing a common baseline in this metric.

- In contrast, the adult mortality rates diverge significantly between developed and developing countries, with developed nations recording a notably lower rate of 229, while developing countries face a higher rate of 723.

- As a result, the average adult mortality rate for developing countries surpasses that of developed countries, more than doubling the latter's rate. This underscores a substantial disparity in adult mortality between the two categories.

In [7]:
%%sql

-- GDP summary statistics by status
SELECT status,
	   COUNT(DISTINCT country) AS country_tally,
	   MIN(gdp) AS min_gdp,
	   MAX(gdp) AS max_gdp,
	   PERCENTILE_CONT(0.5) WITHIN GROUP (ORDER BY gdp) AS median_gdp,
	   ROUND(CAST(AVG(gdp) AS NUMERIC), 1) AS avg_gdp,
	   ROUND(CAST(STDDEV(gdp) AS NUMERIC), 1) AS std_gdp
  FROM world_life_expectancy
 WHERE gdp != 0
 GROUP BY status
;

 * postgresql://postgres:***@localhost:5433/world_life_expectancy
2 rows affected.


[('Developed', 28, 12, 119173, 13561.0, Decimal('22053.5'), Decimal('22870.8')),
 ('Developing', 130, 2, 88565, 1244.0, Decimal('4292.3'), Decimal('8782.2'))]

- On average, the GDP of developed countries surpasses that of developing countries by over five times, despite the fact that there are approximately 80% fewer developing countries in comparison.

- The highest GDP in developing countries is approximately 25% lower than the maximum GDP observed in developed countries.

**NOTE**

Countries with missing values in this variable were excluded from this comparison, including:

Iran (Islamic Republic of), Viet Nam, Syrian Arab Republic, Papua New Guinea, United Kingdom of Great Britain and Northern Ireland, Sao Tome and Principe, Kyrgyzstan, Slovakia, Lao People's Democratic Republic, Saint Lucia, Libya, Congo, Democratic Republic of the Congo, Republic of Korea, Democratic People's Republic of Korea, Venezuela (Bolivarian Republic of), Micronesia (Federated States of), Eritrea, Saint Vincent and the Grenadines, United Republic of Tanzania, Egypt, Bahamas, Gambia, Iraq, United States of America, Côte d'Ivoire, Yemen, South Sudan, The former Yugoslav republic of Macedonia, Bolivia (Plurinational State of), Republic of Moldova, Czechia, Somalia.

(TOTAL: 33)

In [8]:
%%sql

-- Schooling summary statistics by status
SELECT status,
	   COUNT(DISTINCT country) AS country_tally,
	   MIN(schooling) AS min_schooling,
	   MAX(schooling) AS max_schooling,
	   ROUND(CAST(PERCENTILE_CONT(0.5) WITHIN GROUP (ORDER BY schooling) AS NUMERIC), 1) AS median_schooling,
	   ROUND(CAST(AVG(schooling) AS NUMERIC), 1) AS avg_schooling,
	   ROUND(CAST(STDDEV(gdp) AS NUMERIC), 1) AS std_schooling
  FROM world_life_expectancy
 WHERE schooling != 0
 GROUP BY status
;

 * postgresql://postgres:***@localhost:5433/world_life_expectancy
2 rows affected.


[('Developed', 29, 11.5, 20.7, Decimal('15.8'), Decimal('15.8'), Decimal('22830.4')),
 ('Developing', 144, 2.8, 18.3, Decimal('11.7'), Decimal('11.4'), Decimal('8408.0'))]

- Developing countries demonstrate a substantially lower minimum schooling rate in comparison to developed countries, registering figures of 2.8 and 11.5, respectively.

- The maximum schooling rates, while relatively close, slightly favor developed countries with a rate of 20.8, while developing countries follow closely with 18.3.

- On average, developed countries outpace their developing counterparts in schooling rates, boasting a higher average of 15.8 compared to the average of 11.4 observed in developing countries.

**NOTE**

Countries with missing values in this variable were excluded from this comparison, including:

Montenegro, United Kingdom of Great Britain and Northern Ireland, Bosnia and Herzegovina, Democratic Republic of the Congo, Republic of Korea, Democratic People's Republic of Korea, Micronesia (Federated States of), United Republic of Tanzania, Timor-Leste, United States of America, Côte d'Ivoire, South Sudan, Turkmenistan, Republic of Moldova, Equatorial Guinea, Antigua and Barbuda, Czechia, Somalia.

(TOTAL: 18)

# LIFE EXPECTANCY IN DEVELOPED COUNTRIES

In [9]:
%%sql

-- Life expectancy summary statistics in developed countries by region
SELECT region,
	   COUNT(DISTINCT country) AS country_tally,
	   MIN(life_expectancy) AS min_life_expectancy,
	   MAX(life_expectancy) AS max_life_expectancy,
	   ROUND(CAST(PERCENTILE_CONT(0.5) WITHIN GROUP (ORDER BY life_expectancy) AS NUMERIC), 1) AS median_life_expectancy,
	   ROUND(CAST(AVG(life_expectancy) AS NUMERIC), 1) AS avg_life_expectancy,
	   ROUND(CAST(STDDEV(life_expectancy) AS NUMERIC), 1) AS std_life_expectancy
  FROM world_life_expectancy
 WHERE status = 'Developed'
 GROUP BY region
 ORDER BY country_tally
;

 * postgresql://postgres:***@localhost:5433/world_life_expectancy
4 rows affected.


[('Americas', 1, 76.8, 79.3, Decimal('78.1'), Decimal('78.1'), Decimal('0.8')),
 ('Asia', 2, 78.3, 87.0, Decimal('82.1'), Decimal('82.0'), Decimal('1.7')),
 ('Oceania', 2, 78.5, 89.0, Decimal('81.3'), Decimal('81.6'), Decimal('2.3')),
 ('Europe', 27, 69.9, 89.0, Decimal('78.8'), Decimal('78.9'), Decimal('4.1'))]

- Asia boasts the highest life expectancy among developed countries across regions, averaging 82 years over time. It's noteworthy, however, that this statistic primarily reflects the experiences of Japan and Singapore, the only two developed countries in the Asian region.

- In contrast, the Americas exhibit the lowest life expectancy among developed countries, averaging 78.1 years over time. Notably, this region also features the narrowest range of life expectancy. It's important to note that the USA is the sole developed country in the Americas.

- Europe emerges as the most diverse region, showcasing the widest range of life expectancy from 69.9 years to 89 years among developed countries.

In [10]:
%%sql

-- Average life expectancy in developed countries over time
WITH
avg_life_expectancy_cte AS (
SELECT year,
	   ROUND(CAST(AVG(life_expectancy) AS NUMERIC), 1) AS avg_life_expectancy
  FROM world_life_expectancy
 WHERE status = 'Developed'
 GROUP BY year
)

SELECT year,
	   avg_life_expectancy,
	   COALESCE(avg_life_expectancy - LAG(avg_life_expectancy) OVER (ORDER BY year), 0) AS difference_from_previous_year
  FROM avg_life_expectancy_cte
 ORDER BY year
;

 * postgresql://postgres:***@localhost:5433/world_life_expectancy
16 rows affected.


[(2007, Decimal('76.8'), Decimal('0')),
 (2008, Decimal('77.1'), Decimal('0.3')),
 (2009, Decimal('77.5'), Decimal('0.4')),
 (2010, Decimal('77.9'), Decimal('0.4')),
 (2011, Decimal('78.4'), Decimal('0.5')),
 (2012, Decimal('78.6'), Decimal('0.2')),
 (2013, Decimal('79.1'), Decimal('0.5')),
 (2014, Decimal('79.3'), Decimal('0.2')),
 (2015, Decimal('78.9'), Decimal('-0.4')),
 (2016, Decimal('79.6'), Decimal('0.7')),
 (2017, Decimal('80.1'), Decimal('0.5')),
 (2018, Decimal('80.7'), Decimal('0.6')),
 (2019, Decimal('80.4'), Decimal('-0.3')),
 (2020, Decimal('80.7'), Decimal('0.3')),
 (2021, Decimal('81.1'), Decimal('0.4')),
 (2022, Decimal('80.7'), Decimal('-0.4'))]

- Over the span of 2007 to 2022, developed countries have witnessed a cumulative average increase in life expectancy, totaling 3.9 years.

- Specifically, the years 2015, 2019, and 2022 marked slight decreases in the average life expectancy of developed countries, each falling below 0.5 years.

- Notably, the most substantial surge in average life expectancy within developed countries occurred between 2017 and 2018, registering an increase of 0.6 years.


In [11]:
%%sql

-- Top 5 developed countries with highest life expectancy increase over time
SELECT region, country,
	   MIN(life_expectancy) AS min_life_expectancy,
	   MAX(life_expectancy) AS max_life_expectancy,
	   ROUND(CAST(MAX(life_expectancy) - MIN(life_expectancy) AS NUMERIC),1) AS overall_life_increase
  FROM world_life_expectancy
 WHERE status = 'Developed'
 GROUP BY region, country
 ORDER BY overall_life_increase DESC
 LIMIT 5
;

 * postgresql://postgres:***@localhost:5433/world_life_expectancy
5 rows affected.


[('Europe', 'Portugal', 76.6, 89.0, Decimal('12.4')),
 ('Europe', 'Slovenia', 76.0, 88.0, Decimal('12.0')),
 ('Europe', 'Belgium', 77.6, 89.0, Decimal('11.4')),
 ('Europe', 'Germany', 78.0, 89.0, Decimal('11.0')),
 ('Oceania', 'New Zealand', 78.5, 89.0, Decimal('10.5'))]

Portugal has shown the biggest boost in life expectancy among developed countries, going up by an impressive 12.4 years—from 76.6 years to 89 years.

In [12]:
%%sql

-- Top 5 developed countries with lowest life expectancy increase over time
SELECT region, country,
	   MIN(life_expectancy) AS min_life_expectancy,
	   MAX(life_expectancy) AS max_life_expectancy,
	   ROUND(CAST(MAX(life_expectancy) - MIN(life_expectancy) AS NUMERIC),1) AS overall_life_increase
  FROM world_life_expectancy
 WHERE status = 'Developed'
 GROUP BY region, country
 ORDER BY overall_life_increase
 LIMIT 5
;

 * postgresql://postgres:***@localhost:5433/world_life_expectancy
5 rows affected.


[('Americas', 'United States of America', 76.8, 79.3, Decimal('2.5')),
 ('Asia', 'Japan', 81.1, 83.7, Decimal('2.6')),
 ('Europe', 'Croatia', 74.7, 78.0, Decimal('3.3')),
 ('Europe', 'Bulgaria', 71.1, 74.5, Decimal('3.4')),
 ('Europe', 'Slovakia', 73.0, 76.7, Decimal('3.7'))]

On the flip side, the USA had the smallest improvement in life expectancy among developed countries, going up by 2.5 years—from 76.8 years to 79.3 years.

In [13]:
%%sql

-- Average life expectancy in developed countries per region over time
WITH
life_exp_developed AS (
SELECT region, year, 
	   ROUND(CAST(AVG(life_expectancy) AS NUMERIC), 1) AS avg_life_expectancy
  FROM world_life_expectancy
 WHERE status = 'Developed'
 GROUP BY region, year
 ORDER BY region, year
)
	   
SELECT region, year, avg_life_expectancy,
	   COALESCE(avg_life_expectancy - LAG(avg_life_expectancy) OVER (PARTITION BY region
																		 ORDER BY year), 0) AS difference_from_previous_year
FROM life_exp_developed
ORDER BY region, year
;

 * postgresql://postgres:***@localhost:5433/world_life_expectancy
64 rows affected.


[('Americas', 2007, Decimal('76.8'), Decimal('0')),
 ('Americas', 2008, Decimal('76.9'), Decimal('0.1')),
 ('Americas', 2009, Decimal('77.0'), Decimal('0.1')),
 ('Americas', 2010, Decimal('77.2'), Decimal('0.2')),
 ('Americas', 2011, Decimal('77.5'), Decimal('0.3')),
 ('Americas', 2012, Decimal('77.5'), Decimal('0.0')),
 ('Americas', 2013, Decimal('77.8'), Decimal('0.3')),
 ('Americas', 2014, Decimal('78.1'), Decimal('0.3')),
 ('Americas', 2015, Decimal('78.2'), Decimal('0.1')),
 ('Americas', 2016, Decimal('78.5'), Decimal('0.3')),
 ('Americas', 2017, Decimal('78.7'), Decimal('0.2')),
 ('Americas', 2018, Decimal('78.7'), Decimal('0.0')),
 ('Americas', 2019, Decimal('78.8'), Decimal('0.1')),
 ('Americas', 2020, Decimal('78.9'), Decimal('0.1')),
 ('Americas', 2021, Decimal('79.1'), Decimal('0.2')),
 ('Americas', 2022, Decimal('79.3'), Decimal('0.2')),
 ('Asia', 2007, Decimal('79.7'), Decimal('0')),
 ('Asia', 2008, Decimal('80.1'), Decimal('0.4')),
 ('Asia', 2009, Decimal('80.4'), Decimal('0.3')),
 ('Asia', 2010, Decimal('80.6'), Decimal('0.2')),
 ('Asia', 2011, Decimal('80.9'), Decimal('0.3')),
 ('Asia', 2012, Decimal('82.0'), Decimal('1.1')),
 ('Asia', 2013, Decimal('84.7'), Decimal('2.7')),
 ('Asia', 2014, Decimal('81.8'), Decimal('-2.9')),
 ('Asia', 2015, Decimal('82.0'), Decimal('0.2')),
 ('Asia', 2016, Decimal('82.3'), Decimal('0.3')),
 ('Asia', 2017, Decimal('82.5'), Decimal('0.2')),
 ('Asia', 2018, Decimal('82.3'), Decimal('-0.2')),
 ('Asia', 2019, Decimal('82.9'), Decimal('0.6')),
 ('Asia', 2020, Decimal('83.1'), Decimal('0.2')),
 ('Asia', 2021, Decimal('83.2'), Decimal('0.1')),
 ('Asia', 2022, Decimal('83.4'), Decimal('0.2')),
 ('Europe', 2007, Decimal('76.4'), Decimal('0')),
 ('Europe', 2008, Decimal('76.8'), Decimal('0.4')),
 ('Europe', 2009, Decimal('77.2'), Decimal('0.4')),
 ('Europe', 2010, Decimal('77.5'), Decimal('0.3')),
 ('Europe', 2011, Decimal('77.9'), Decimal('0.4')),
 ('Europe', 2012, Decimal('78.2'), Decimal('0.3')),
 ('Europe', 2013, Decimal('78.7'), Decimal('0.5')),
 ('Europe', 2014, Decimal('79.0'), Decimal('0.3')),
 ('Europe', 2015, Decimal('78.6'), Decimal('-0.4')),
 ('Europe', 2016, Decimal('79.1'), Decimal('0.5')),
 ('Europe', 2017, Decimal('79.6'), Decimal('0.5')),
 ('Europe', 2018, Decimal('80.4'), Decimal('0.8')),
 ('Europe', 2019, Decimal('80.2'), Decimal('-0.2')),
 ('Europe', 2020, Decimal('80.5'), Decimal('0.3')),
 ('Europe', 2021, Decimal('81.0'), Decimal('0.5')),
 ('Europe', 2022, Decimal('80.5'), Decimal('-0.5')),
 ('Oceania', 2007, Decimal('79.0'), Decimal('0')),
 ('Oceania', 2008, Decimal('79.2'), Decimal('0.2')),
 ('Oceania', 2009, Decimal('79.3'), Decimal('0.1')),
 ('Oceania', 2010, Decimal('81.0'), Decimal('1.7')),
 ('Oceania', 2011, Decimal('82.6'), Decimal('1.6')),
 ('Oceania', 2012, Decimal('80.5'), Decimal('-2.1')),
 ('Oceania', 2013, Decimal('80.5'), Decimal('0.0')),
 ('Oceania', 2014, Decimal('81.2'), Decimal('0.7')),
 ('Oceania', 2015, Decimal('81.2'), Decimal('0.0')),
 ('Oceania', 2016, Decimal('83.3'), Decimal('2.1')),
 ('Oceania', 2017, Decimal('85.5'), Decimal('2.2')),
 ('Oceania', 2018, Decimal('84.0'), Decimal('-1.5')),
 ('Oceania', 2019, Decimal('81.7'), Decimal('-2.3')),
 ('Oceania', 2020, Decimal('81.9'), Decimal('0.2')),
 ('Oceania', 2021, Decimal('82.1'), Decimal('0.2')),
 ('Oceania', 2022, Decimal('82.2'), Decimal('0.1'))]

# MORTALITY IN DEVELOPED COUNTRIES

In [14]:
%%sql

-- Mortality summary statistics in developed countries by region
SELECT region,
	   COUNT(DISTINCT country) AS country_tally,
	   MIN(adult_mortality) AS min_adult_mortality,
	   MAX(adult_mortality) AS max_adult_mortality,
	   PERCENTILE_CONT(0.5) WITHIN GROUP (ORDER BY adult_mortality) AS median_adult_mortality,
	   ROUND(CAST(AVG(adult_mortality) AS NUMERIC), 1) AS avg_adult_mortality,
	   ROUND(CAST(STDDEV(adult_mortality) AS NUMERIC), 1) AS std_adult_mortality
  FROM world_life_expectancy
 WHERE status = 'Developed'
 GROUP BY region
 ORDER BY country_tally
;

 * postgresql://postgres:***@localhost:5433/world_life_expectancy
4 rows affected.


[('Americas', 1, 11, 115, 18.0, Decimal('58.2'), Decimal('50.4')),
 ('Asia', 2, 6, 78, 64.0, Decimal('59.6'), Decimal('18.4')),
 ('Oceania', 2, 6, 87, 70.0, Decimal('67.3'), Decimal('17.4')),
 ('Europe', 27, 1, 229, 75.0, Decimal('82.9'), Decimal('50.1'))]

- Europe consistently exhibits the highest average mortality rate and the widest range of mortality rates among all regions over time within developed countries.

- In contrast, the Americas region, solely represented by the USA in this analysis, boasts the lowest mortality rate across all regions over time, approximately 30% lower than the average mortality rate in Europe.


In [15]:
%%sql

-- Total mortality in developed countries over time
WITH
total_mortality_cte AS (
SELECT year,
	   SUM(adult_mortality) AS total_mortality
  FROM world_life_expectancy
 WHERE status = 'Developed'
 GROUP BY year
)

SELECT year,
	   total_mortality,
	   COALESCE(total_mortality - LAG(total_mortality) OVER (ORDER BY year), 0) AS difference_from_previous_year
  FROM total_mortality_cte
 ORDER BY year
;

 * postgresql://postgres:***@localhost:5433/world_life_expectancy
16 rows affected.


[(2007, 2935, 0),
 (2008, 2889, -46),
 (2009, 2767, -122),
 (2010, 2622, -145),
 (2011, 2703, 81),
 (2012, 2867, 164),
 (2013, 2993, 126),
 (2014, 2429, -564),
 (2015, 2466, 37),
 (2016, 2456, -10),
 (2017, 2385, -71),
 (2018, 2102, -283),
 (2019, 2275, 173),
 (2020, 2133, -142),
 (2021, 2381, 248),
 (2022, 2396, 15)]

- The most substantial reduction in mortality occurred between 2013 and 2014, declining from 2993 to 2429 in developed countries.

- Conversely, the most notable increase in mortality transpired between 2020 and 2021, rising from 2133 to 2381 in developed countries.

In [16]:
%%sql

-- Top 5 developed countries with highest mortality increase over time
SELECT region, country,
	   MIN(adult_mortality) AS min_mortality,
	   MAX(adult_mortality) AS max_mortality,
	   ROUND(CAST(MAX(adult_mortality) - MIN(adult_mortality) AS NUMERIC),1) AS overall_mortality_increase
  FROM world_life_expectancy
 WHERE status = 'Developed'
 GROUP BY region, country
 ORDER BY overall_mortality_increase DESC
 LIMIT 5
;

 * postgresql://postgres:***@localhost:5433/world_life_expectancy
5 rows affected.


[('Europe', 'Lithuania', 2, 229, Decimal('227.0')),
 ('Europe', 'Latvia', 18, 229, Decimal('211.0')),
 ('Europe', 'Hungary', 15, 193, Decimal('178.0')),
 ('Europe', 'Bulgaria', 14, 163, Decimal('149.0')),
 ('Europe', 'Poland', 12, 153, Decimal('141.0'))]

Lithuania witnessed the most substantial overall increase in mortality over time, escalating from 2 to 229.

In [17]:
%%sql

-- Top 5 developed countries with lowest mortality increase over time
SELECT region, country,
	   MIN(adult_mortality) AS min_mortality,
	   MAX(adult_mortality) AS max_mortality,
	   ROUND(CAST(MAX(adult_mortality) - MIN(adult_mortality) AS NUMERIC),1) AS overall_mortality_increase
  FROM world_life_expectancy
 WHERE status = 'Developed'
 GROUP BY region, country
 ORDER BY overall_mortality_increase
 LIMIT 5
;

 * postgresql://postgres:***@localhost:5433/world_life_expectancy
5 rows affected.


[('Europe', 'Romania', 133, 179, Decimal('46.0')),
 ('Europe', 'Cyprus', 6, 69, Decimal('63.0')),
 ('Europe', 'Sweden', 7, 73, Decimal('66.0')),
 ('Asia', 'Japan', 7, 74, Decimal('67.0')),
 ('Europe', 'Iceland', 5, 74, Decimal('69.0'))]

In contrast, Romania registered the smallest overall increase in mortality over time, rising from 133 to 179.

In [18]:
%%sql

-- Rolling total of mortality in developed countries per region over time
WITH
life_exp_developed AS (
SELECT region, year, 
	   SUM(adult_mortality) AS adult_mortality
  FROM world_life_expectancy
 WHERE status = 'Developed'
 GROUP BY region, year
 ORDER BY region, year
)
	   
SELECT region, year, adult_mortality AS total_mortality,
	   SUM(adult_mortality) OVER(PARTITION BY region
								 ORDER BY year) AS mortality_rolling_total
FROM life_exp_developed
;

 * postgresql://postgres:***@localhost:5433/world_life_expectancy
64 rows affected.


[('Americas', 2007, 114, Decimal('114')),
 ('Americas', 2008, 115, Decimal('229')),
 ('Americas', 2009, 115, Decimal('344')),
 ('Americas', 2010, 114, Decimal('458')),
 ('Americas', 2011, 111, Decimal('569')),
 ('Americas', 2012, 112, Decimal('681')),
 ('Americas', 2013, 113, Decimal('794')),
 ('Americas', 2014, 11, Decimal('805')),
 ('Americas', 2015, 18, Decimal('823')),
 ('Americas', 2016, 18, Decimal('841')),
 ('Americas', 2017, 15, Decimal('856')),
 ('Americas', 2018, 16, Decimal('872')),
 ('Americas', 2019, 16, Decimal('888')),
 ('Americas', 2020, 16, Decimal('904')),
 ('Americas', 2021, 14, Decimal('918')),
 ('Americas', 2022, 13, Decimal('931')),
 ('Asia', 2007, 152, Decimal('152')),
 ('Asia', 2008, 148, Decimal('300')),
 ('Asia', 2009, 81, Decimal('381')),
 ('Asia', 2010, 80, Decimal('461')),
 ('Asia', 2011, 139, Decimal('600')),
 ('Asia', 2012, 138, Decimal('738')),
 ('Asia', 2013, 134, Decimal('872')),
 ('Asia', 2014, 132, Decimal('1004')),
 ('Asia', 2015, 130, Decimal('1134')),
 ('Asia', 2016, 126, Decimal('1260')),
 ('Asia', 2017, 123, Decimal('1383')),
 ('Asia', 2018, 70, Decimal('1453')),
 ('Asia', 2019, 117, Decimal('1570')),
 ('Asia', 2020, 113, Decimal('1683')),
 ('Asia', 2021, 113, Decimal('1796')),
 ('Asia', 2022, 110, Decimal('1906')),
 ('Europe', 2007, 2504, Decimal('2504')),
 ('Europe', 2008, 2468, Decimal('4972')),
 ('Europe', 2009, 2415, Decimal('7387')),
 ('Europe', 2010, 2275, Decimal('9662')),
 ('Europe', 2011, 2305, Decimal('11967')),
 ('Europe', 2012, 2472, Decimal('14439')),
 ('Europe', 2013, 2604, Decimal('17043')),
 ('Europe', 2014, 2144, Decimal('19187')),
 ('Europe', 2015, 2177, Decimal('21364')),
 ('Europe', 2016, 2173, Decimal('23537')),
 ('Europe', 2017, 2112, Decimal('25649')),
 ('Europe', 2018, 1881, Decimal('27530')),
 ('Europe', 2019, 2074, Decimal('29604')),
 ('Europe', 2020, 1874, Decimal('31478')),
 ('Europe', 2021, 2181, Decimal('33659')),
 ('Europe', 2022, 2148, Decimal('35807')),
 ('Oceania', 2007, 165, Decimal('165')),
 ('Oceania', 2008, 158, Decimal('323')),
 ('Oceania', 2009, 156, Decimal('479')),
 ('Oceania', 2010, 153, Decimal('632')),
 ('Oceania', 2011, 148, Decimal('780')),
 ('Oceania', 2012, 145, Decimal('925')),
 ('Oceania', 2013, 142, Decimal('1067')),
 ('Oceania', 2014, 142, Decimal('1209')),
 ('Oceania', 2015, 141, Decimal('1350')),
 ('Oceania', 2016, 139, Decimal('1489')),
 ('Oceania', 2017, 135, Decimal('1624')),
 ('Oceania', 2018, 135, Decimal('1759')),
 ('Oceania', 2019, 68, Decimal('1827')),
 ('Oceania', 2020, 130, Decimal('1957')),
 ('Oceania', 2021, 73, Decimal('2030')),
 ('Oceania', 2022, 125, Decimal('2155'))]

# LIFE EXPECTANCY IN DEVELOPING COUNTRIES

In [19]:
%%sql

-- Life expectancy summary statistics in developing countries by region
SELECT region,
	   COUNT(DISTINCT country) AS country_tally,
	   MIN(life_expectancy) AS min_life_expectancy,
	   MAX(life_expectancy) AS max_life_expectancy,
	   ROUND(CAST(PERCENTILE_CONT(0.5) WITHIN GROUP (ORDER BY life_expectancy) AS NUMERIC), 1) AS median_life_expectancy,
	   ROUND(CAST(AVG(life_expectancy) AS NUMERIC), 1) AS avg_life_expectancy,
	   ROUND(CAST(STDDEV(life_expectancy) AS NUMERIC), 1) AS std_life_expectancy
  FROM world_life_expectancy
 WHERE status = 'Developing'
 GROUP BY region
 ORDER BY country_tally
;

 * postgresql://postgres:***@localhost:5433/world_life_expectancy
5 rows affected.


[('Oceania', 8, 58.9, 79.0, Decimal('68.7'), Decimal('68.6'), Decimal('4.0')),
 ('Europe', 12, 64.6, 89.0, Decimal('74.9'), Decimal('74.7'), Decimal('5.3')),
 ('Americas', 32, 36.3, 87.0, Decimal('73.8'), Decimal('73.3'), Decimal('4.4')),
 ('Asia', 45, 54.8, 87.0, Decimal('72.0'), Decimal('70.6'), Decimal('5.4')),
 ('Africa', 54, 39.0, 79.0, Decimal('57.8'), Decimal('58.6'), Decimal('8.0'))]

- Europe stands out with the highest average, minimum, and maximum life expectancy among developing countries across all regions.

- In the Americas, the range of life expectancy in developing countries is the widest across all regions, with the minimum value even lower than Africa, which has the lowest average life expectancy.

In [20]:
%%sql

-- Average life expectancy in developing countries over time
WITH
avg_life_expectancy_cte AS (
SELECT year,
	   ROUND(CAST(AVG(life_expectancy) AS NUMERIC), 1) AS avg_life_expectancy
  FROM world_life_expectancy
 WHERE status = 'Developing'
 GROUP BY year
)

SELECT year,
	   avg_life_expectancy,
	   COALESCE(avg_life_expectancy - LAG(avg_life_expectancy) OVER (ORDER BY year), 0) AS difference_from_previous_year
  FROM avg_life_expectancy_cte
 ORDER BY year
;

 * postgresql://postgres:***@localhost:5433/world_life_expectancy
16 rows affected.


[(2007, Decimal('64.6'), Decimal('0')),
 (2008, Decimal('65.0'), Decimal('0.4')),
 (2009, Decimal('65.2'), Decimal('0.2')),
 (2010, Decimal('65.2'), Decimal('0.0')),
 (2011, Decimal('65.4'), Decimal('0.2')),
 (2012, Decimal('66.0'), Decimal('0.6')),
 (2013, Decimal('66.5'), Decimal('0.5')),
 (2014, Decimal('66.9'), Decimal('0.4')),
 (2015, Decimal('67.4'), Decimal('0.5')),
 (2016, Decimal('67.9'), Decimal('0.5')),
 (2017, Decimal('67.9'), Decimal('0.0')),
 (2018, Decimal('68.5'), Decimal('0.6')),
 (2019, Decimal('68.9'), Decimal('0.4')),
 (2020, Decimal('69.2'), Decimal('0.3')),
 (2021, Decimal('69.5'), Decimal('0.3')),
 (2022, Decimal('69.7'), Decimal('0.2'))]

- Over the period spanning from 2007 to 2022, there has been a notable collective increase in the average life expectancy of developing countries, amounting to 5.1 years.

- Interestingly, there was a notable plateau in the average life expectancy of developing countries between the periods of 2009-2010 and 2016-2017, wherein no discernible increase was observed.

- Specifically, the years 2012 and 2018 stand out as having experienced the most substantial spikes in the average life expectancy of developing countries, both witnessing an increase of 0.6 years.

In [21]:
%%sql

-- Top 5 developing countries with highest life expectancy increase over time
SELECT region, country,
	   MIN(life_expectancy) AS min_life_expectancy,
	   MAX(life_expectancy) AS max_life_expectancy,
	   ROUND(CAST(MAX(life_expectancy) - MIN(life_expectancy) AS NUMERIC),1) AS life_increase_15_years
  FROM world_life_expectancy
 WHERE status = 'Developing'
 GROUP BY region, country
 ORDER BY life_increase_15_years DESC
 LIMIT 5
;

 * postgresql://postgres:***@localhost:5433/world_life_expectancy
5 rows affected.


[('Americas', 'Haiti', 36.3, 65.0, Decimal('28.7')),
 ('Africa', 'Zimbabwe', 44.3, 67.0, Decimal('22.7')),
 ('Africa', 'Eritrea', 45.3, 67.0, Decimal('21.7')),
 ('Africa', 'Uganda', 46.6, 67.0, Decimal('20.4')),
 ('Africa', 'Rwanda', 48.3, 68.0, Decimal('19.7'))]

Haiti stands out with a remarkable overall increase in life expectancy, surging by 28.7 years over 16 years—from 36.3 years to 65 years. This represents the most significant improvement among all developing countries.

In [22]:
%%sql

-- Top 5 developing countries with lowest life expectancy increase over time
SELECT region, country,
	   MIN(life_expectancy) AS min_life_expectancy,
	   MAX(life_expectancy) AS max_life_expectancy,
	   ROUND(CAST(MAX(life_expectancy) - MIN(life_expectancy) AS NUMERIC),1) AS life_increase_15_years
  FROM world_life_expectancy
 WHERE status = 'Developing'
 GROUP BY region, country
 ORDER BY life_increase_15_years
 LIMIT 5
;

 * postgresql://postgres:***@localhost:5433/world_life_expectancy
5 rows affected.


[('Americas', 'Guyana', 65.0, 66.3, Decimal('1.3')),
 ('Africa', 'Seychelles', 71.8, 73.2, Decimal('1.4')),
 ('Asia', 'Kuwait', 73.2, 74.7, Decimal('1.5')),
 ('Americas', 'Venezuela (Bolivarian Republic of)', 72.4, 74.1, Decimal('1.7')),
 ('Asia', 'Philippines', 66.8, 68.5, Decimal('1.7'))]

In contrast, Guyana experienced the smallest overall increase in life expectancy, with a modest gain of only 1.3 years. Over the same 16-year period, Guyana's life expectancy inched up from 65 years to 66.3 years.

In [23]:
%%sql

-- Average life expectancy in developing countries per region over time
WITH
life_exp_developed AS (
SELECT region, year, 
	   ROUND(CAST(AVG(life_expectancy) AS NUMERIC), 1) AS avg_life_expectancy
  FROM world_life_expectancy
 WHERE status = 'Developing'
 GROUP BY region, year
 ORDER BY region, year
)
	   
SELECT region, year, avg_life_expectancy,
	   COALESCE(avg_life_expectancy - LAG(avg_life_expectancy) OVER (PARTITION BY region
																		 ORDER BY year), 0) AS difference_from_previous_year
FROM life_exp_developed
ORDER BY region, year
;

 * postgresql://postgres:***@localhost:5433/world_life_expectancy
80 rows affected.


[('Africa', 2007, Decimal('54.7'), Decimal('0')),
 ('Africa', 2008, Decimal('55.2'), Decimal('0.5')),
 ('Africa', 2009, Decimal('55.6'), Decimal('0.4')),
 ('Africa', 2010, Decimal('55.6'), Decimal('0.0')),
 ('Africa', 2011, Decimal('55.9'), Decimal('0.3')),
 ('Africa', 2012, Decimal('57.1'), Decimal('1.2')),
 ('Africa', 2013, Decimal('57.4'), Decimal('0.3')),
 ('Africa', 2014, Decimal('58.1'), Decimal('0.7')),
 ('Africa', 2015, Decimal('59.1'), Decimal('1.0')),
 ('Africa', 2016, Decimal('59.8'), Decimal('0.7')),
 ('Africa', 2017, Decimal('60.1'), Decimal('0.3')),
 ('Africa', 2018, Decimal('60.6'), Decimal('0.5')),
 ('Africa', 2019, Decimal('61.4'), Decimal('0.8')),
 ('Africa', 2020, Decimal('62.3'), Decimal('0.9')),
 ('Africa', 2021, Decimal('62.3'), Decimal('0.0')),
 ('Africa', 2022, Decimal('62.7'), Decimal('0.4')),
 ('Americas', 2007, Decimal('72.3'), Decimal('0')),
 ('Americas', 2008, Decimal('72.1'), Decimal('-0.2')),
 ('Americas', 2009, Decimal('72.7'), Decimal('0.6')),
 ('Americas', 2010, Decimal('72.2'), Decimal('-0.5')),
 ('Americas', 2011, Decimal('72.2'), Decimal('0.0')),
 ('Americas', 2012, Decimal('72.6'), Decimal('0.4')),
 ('Americas', 2013, Decimal('73.2'), Decimal('0.6')),
 ('Americas', 2014, Decimal('73.4'), Decimal('0.2')),
 ('Americas', 2015, Decimal('73.8'), Decimal('0.4')),
 ('Americas', 2016, Decimal('73.7'), Decimal('-0.1')),
 ('Americas', 2017, Decimal('73.1'), Decimal('-0.6')),
 ('Americas', 2018, Decimal('74.2'), Decimal('1.1')),
 ('Americas', 2019, Decimal('74.2'), Decimal('0.0')),
 ('Americas', 2020, Decimal('74.2'), Decimal('0.0')),
 ('Americas', 2021, Decimal('74.6'), Decimal('0.4')),
 ('Americas', 2022, Decimal('75.0'), Decimal('0.4')),
 ('Asia', 2007, Decimal('68.4'), Decimal('0')),
 ('Asia', 2008, Decimal('69.1'), Decimal('0.7')),
 ('Asia', 2009, Decimal('69.0'), Decimal('-0.1')),
 ('Asia', 2010, Decimal('69.3'), Decimal('0.3')),
 ('Asia', 2011, Decimal('69.4'), Decimal('0.1')),
 ('Asia', 2012, Decimal('69.7'), Decimal('0.3')),
 ('Asia', 2013, Decimal('70.1'), Decimal('0.4')),
 ('Asia', 2014, Decimal('70.5'), Decimal('0.4')),
 ('Asia', 2015, Decimal('70.7'), Decimal('0.2')),
 ('Asia', 2016, Decimal('71.3'), Decimal('0.6')),
 ('Asia', 2017, Decimal('71.5'), Decimal('0.2')),
 ('Asia', 2018, Decimal('71.7'), Decimal('0.2')),
 ('Asia', 2019, Decimal('71.9'), Decimal('0.2')),
 ('Asia', 2020, Decimal('72.0'), Decimal('0.1')),
 ('Asia', 2021, Decimal('72.3'), Decimal('0.3')),
 ('Asia', 2022, Decimal('72.5'), Decimal('0.2')),
 ('Europe', 2007, Decimal('72.7'), Decimal('0')),
 ('Europe', 2008, Decimal('73.1'), Decimal('0.4')),
 ('Europe', 2009, Decimal('72.5'), Decimal('-0.6')),
 ('Europe', 2010, Decimal('72.6'), Decimal('0.1')),
 ('Europe', 2011, Decimal('73.0'), Decimal('0.4')),
 ('Europe', 2012, Decimal('72.9'), Decimal('-0.1')),
 ('Europe', 2013, Decimal('73.8'), Decimal('0.9')),
 ('Europe', 2014, Decimal('74.4'), Decimal('0.6')),
 ('Europe', 2015, Decimal('74.7'), Decimal('0.3')),
 ('Europe', 2016, Decimal('74.4'), Decimal('-0.3')),
 ('Europe', 2017, Decimal('75.2'), Decimal('0.8')),
 ('Europe', 2018, Decimal('76.9'), Decimal('1.7')),
 ('Europe', 2019, Decimal('77.3'), Decimal('0.4')),
 ('Europe', 2020, Decimal('76.8'), Decimal('-0.5')),
 ('Europe', 2021, Decimal('78.1'), Decimal('1.3')),
 ('Europe', 2022, Decimal('76.6'), Decimal('-1.5')),
 ('Oceania', 2007, Decimal('67.0'), Decimal('0')),
 ('Oceania', 2008, Decimal('67.6'), Decimal('0.6')),
 ('Oceania', 2009, Decimal('67.7'), Decimal('0.1')),
 ('Oceania', 2010, Decimal('68.4'), Decimal('0.7')),
 ('Oceania', 2011, Decimal('67.6'), Decimal('-0.8')),
 ('Oceania', 2012, Decimal('68.3'), Decimal('0.7')),
 ('Oceania', 2013, Decimal('69.1'), Decimal('0.8')),
 ('Oceania', 2014, Decimal('68.5'), Decimal('-0.6')),
 ('Oceania', 2015, Decimal('69.0'), Decimal('0.5')),
 ('Oceania', 2016, Decimal('69.8'), Decimal('0.8')),
 ('Oceania', 2017, Decimal('68.7'), Decimal('-1.1')),
 ('Oceania', 2018, Decimal('68.9'), Decimal('0.2')),
 ('Oceania', 2019, Decimal('69.1'

# MORTALITY IN DEVELOPED COUNTRIES

In [24]:
%%sql

-- Mortality summary statistics in developing countries by region
SELECT region,
	   COUNT(DISTINCT country) AS country_tally,
	   MIN(adult_mortality) AS min_adult_mortality,
	   MAX(adult_mortality) AS max_adult_mortality,
	   PERCENTILE_CONT(0.5) WITHIN GROUP (ORDER BY adult_mortality) AS median_adult_mortality,
	   ROUND(CAST(AVG(adult_mortality) AS NUMERIC), 1) AS avg_adult_mortality,
	   ROUND(CAST(STDDEV(adult_mortality) AS NUMERIC), 1) AS std_adult_mortality
  FROM world_life_expectancy
 WHERE status = 'Developing'
 GROUP BY region
 ORDER BY country_tally
;

 * postgresql://postgres:***@localhost:5433/world_life_expectancy
5 rows affected.


[('Oceania', 8, 2, 335, 165.5, Decimal('152.0'), Decimal('85.4')),
 ('Europe', 12, 1, 327, 120.0, Decimal('131.5'), Decimal('84.8')),
 ('Americas', 32, 1, 682, 146.5, Decimal('133.1'), Decimal('67.5')),
 ('Asia', 45, 1, 321, 135.5, Decimal('136.9'), Decimal('77.4')),
 ('Africa', 54, 1, 723, 278.0, Decimal('266.6'), Decimal('156.5'))]

Africa exhibits the widest range of mortality rates and maintains the highest average mortality over time, distinguishing itself from the other regions, which demonstrate relatively similar patterns.

In [25]:
%%sql

-- Total mortality in developing countries over time
WITH
total_mortality_cte AS (
SELECT year,
	   SUM(adult_mortality) AS total_mortality
  FROM world_life_expectancy
 WHERE status = 'Developing'
 GROUP BY year
)

SELECT year,
	   total_mortality,
	   COALESCE(total_mortality - LAG(total_mortality) OVER (ORDER BY year), 0) AS difference_from_previous_year
  FROM total_mortality_cte
 ORDER BY year
;

 * postgresql://postgres:***@localhost:5433/world_life_expectancy
16 rows affected.


[(2007, 30275, 0),
 (2008, 29167, -1108),
 (2009, 28606, -561),
 (2010, 28329, -277),
 (2011, 31383, 3054),
 (2012, 27810, -3573),
 (2013, 28191, 381),
 (2014, 26816, -1375),
 (2015, 29471, 2655),
 (2016, 27445, -2026),
 (2017, 27236, -209),
 (2018, 26544, -692),
 (2019, 24954, -1590),
 (2020, 25091, 137),
 (2021, 24829, -262),
 (2022, 25578, 749)]

- The most significant surge in total mortality in developing countries is noted between 2010 and 2011, followed by the most substantial decline in the subsequent year, between 2011 and 2012.

- The overall trend in total mortality across developing countries remains relatively stable, fluctuating within the range of approximately 25,000 to 30,000 over time.

In [26]:
%%sql

-- Top 5 developed countries with highest mortality increase over time
SELECT region, country,
	   MIN(adult_mortality) AS min_mortality,
	   MAX(adult_mortality) AS max_mortality,
	   ROUND(CAST(MAX(adult_mortality) - MIN(adult_mortality) AS NUMERIC),1) AS overall_mortality_increase
  FROM world_life_expectancy
 WHERE status = 'Developing'
 GROUP BY region, country
 ORDER BY overall_mortality_increase DESC
 LIMIT 5
;

 * postgresql://postgres:***@localhost:5433/world_life_expectancy
5 rows affected.


[('Africa', 'Zimbabwe', 7, 723, Decimal('716.0')),
 ('Africa', 'Botswana', 3, 699, Decimal('696.0')),
 ('Americas', 'Haiti', 3, 682, Decimal('679.0')),
 ('Africa', 'Lesotho', 52, 675, Decimal('623.0')),
 ('Africa', 'Swaziland', 6, 587, Decimal('581.0'))]

Zimbabwe stands out among developing countries, exhibiting the most substantial overall rise in mortality, increasing from 7 to 723 over time.

In [27]:
%%sql

-- Top 5 developed countries with lowest mortality increase over time
SELECT region, country,
	   MIN(adult_mortality) AS min_mortality,
	   MAX(adult_mortality) AS max_mortality,
	   ROUND(CAST(MAX(adult_mortality) - MIN(adult_mortality) AS NUMERIC),1) AS overall_mortality_increase
  FROM world_life_expectancy
 WHERE status = 'Developing'
 GROUP BY region, country
 ORDER BY overall_mortality_increase
 LIMIT 5
;

 * postgresql://postgres:***@localhost:5433/world_life_expectancy
5 rows affected.


[('Asia', 'Uzbekistan', 183, 192, Decimal('9.0')),
 ('Asia', 'Philippines', 211, 223, Decimal('12.0')),
 ('Americas', 'Venezuela (Bolivarian Republic of)', 157, 172, Decimal('15.0')),
 ('Africa', 'Egypt', 159, 178, Decimal('19.0')),
 ('Americas', 'Uruguay', 111, 131, Decimal('20.0'))]

In contrast, Uzbekistan stands out as the developing country with the minimal overall increase in mortality, moving from 183 to 192 over time.

In [28]:
%%sql

-- Rolling total of mortality in developing countries per region over time
WITH
life_exp_developed AS (
SELECT region, year, 
	   SUM(adult_mortality) AS adult_mortality
  FROM world_life_expectancy
 WHERE status = 'Developing'
 GROUP BY region, year
 ORDER BY region, year
)
	   
SELECT region, year, adult_mortality AS total_mortality,
	   SUM(adult_mortality) OVER(PARTITION BY region
								     ORDER BY year) AS mortality_rolling_total
FROM life_exp_developed
;

 * postgresql://postgres:***@localhost:5433/world_life_expectancy
80 rows affected.


[('Africa', 2007, 14946, Decimal('14946')),
 ('Africa', 2008, 14965, Decimal('29911')),
 ('Africa', 2009, 14901, Decimal('44812')),
 ('Africa', 2010, 14546, Decimal('59358')),
 ('Africa', 2011, 16937, Decimal('76295')),
 ('Africa', 2012, 14815, Decimal('91110')),
 ('Africa', 2013, 15036, Decimal('106146')),
 ('Africa', 2014, 14280, Decimal('120426')),
 ('Africa', 2015, 15470, Decimal('135896')),
 ('Africa', 2016, 14408, Decimal('150304')),
 ('Africa', 2017, 14100, Decimal('164404')),
 ('Africa', 2018, 14074, Decimal('178478')),
 ('Africa', 2019, 13273, Decimal('191751')),
 ('Africa', 2020, 13209, Decimal('204960')),
 ('Africa', 2021, 12771, Decimal('217731')),
 ('Africa', 2022, 12587, Decimal('230318')),
 ('Americas', 2007, 5152, Decimal('5152')),
 ('Americas', 2008, 4598, Decimal('9750')),
 ('Americas', 2009, 4418, Decimal('14168')),
 ('Americas', 2010, 3729, Decimal('17897')),
 ('Americas', 2011, 4250, Decimal('22147')),
 ('Americas', 2012, 4189, Decimal('26336')),
 ('Americas', 2013, 3974, Decimal('30310')),
 ('Americas', 2014, 4040, Decimal('34350')),
 ('Americas', 2015, 4673, Decimal('39023')),
 ('Americas', 2016, 3868, Decimal('42891')),
 ('Americas', 2017, 4413, Decimal('47304')),
 ('Americas', 2018, 4013, Decimal('51317')),
 ('Americas', 2019, 4232, Decimal('55549')),
 ('Americas', 2020, 4185, Decimal('59734')),
 ('Americas', 2021, 4368, Decimal('64102')),
 ('Americas', 2022, 4054, Decimal('68156')),
 ('Asia', 2007, 7267, Decimal('7267')),
 ('Asia', 2008, 6419, Decimal('13686')),
 ('Asia', 2009, 6421, Decimal('20107')),
 ('Asia', 2010, 6936, Decimal('27043')),
 ('Asia', 2011, 6934, Decimal('33977')),
 ('Asia', 2012, 6170, Decimal('40147')),
 ('Asia', 2013, 6365, Decimal('46512')),
 ('Asia', 2014, 5646, Decimal('52158')),
 ('Asia', 2015, 6660, Decimal('58818')),
 ('Asia', 2016, 6213, Decimal('65031')),
 ('Asia', 2017, 5893, Decimal('70924')),
 ('Asia', 2018, 5822, Decimal('76746')),
 ('Asia', 2019, 4860, Decimal('81606')),
 ('Asia', 2020, 5243, Decimal('86849')),
 ('Asia', 2021, 5449, Decimal('92298')),
 ('Asia', 2022, 6248, Decimal('98546')),
 ('Europe', 2007, 1518, Decimal('1518')),
 ('Europe', 2008, 1493, Decimal('3011')),
 ('Europe', 2009, 1752, Decimal('4763')),
 ('Europe', 2010, 1832, Decimal('6595')),
 ('Europe', 2011, 1826, Decimal('8421')),
 ('Europe', 2012, 1774, Decimal('10195')),
 ('Europe', 2013, 1454, Decimal('11649')),
 ('Europe', 2014, 1686, Decimal('13335')),
 ('Europe', 2015, 1781, Decimal('15116')),
 ('Europe', 2016, 1785, Decimal('16901')),
 ('Europe', 2017, 1725, Decimal('18626')),
 ('Europe', 2018, 1478, Decimal('20104')),
 ('Europe', 2019, 1322, Decimal('21426')),
 ('Europe', 2020, 1208, Decimal('22634')),
 ('Europe', 2021, 1198, Decimal('23832')),
 ('Europe', 2022, 1414, Decimal('25246')),
 ('Oceania', 2007, 1392, Decimal('1392')),
 ('Oceania', 2008, 1692, Decimal('3084')),
 ('Oceania', 2009, 1114, Decimal('4198')),
 ('Oceania', 2010, 1286, Decimal('5484')),
 ('Oceania', 2011, 1436, Decimal('6920')),
 ('Oceania', 2012, 862, Decimal('7782')),
 ('Oceania', 2013, 1362, Decimal('9144')),
 ('Oceania', 2014, 1164, Decimal('10308')),
 ('Oceania', 2015, 887, Decimal('11195')),
 ('Oceania', 2016, 1171, Decimal('12366')),
 ('Oceania', 2017, 1105, Decimal('13471')),
 ('Oceania', 2018, 1157, Decimal('14628')),
 ('Oceania', 2019, 1267, Decimal('15895')),
 ('Oceania', 2020, 1246, Decimal('17141')),
 ('Oceania', 2021, 1043, Decimal('18184')),
 ('Oceania', 2022, 1275, Decimal('19459'))]

# SCHOOLING AND GDP'S IMPACT ON LIFE EXPECTANCY AND MORTALITY

In [29]:
%%sql

-- Overal median GDP and schooling per life expectancy quartile by region
WITH
life_exp_bins AS ( 
SELECT region, country, ROUND(CAST(PERCENTILE_CONT(0.5) WITHIN GROUP(ORDER BY life_expectancy) AS NUMERIC), 1) AS median_life_exp,
	   CASE
	   WHEN PERCENT_RANK() OVER(PARTITION BY region
								ORDER BY ROUND(CAST(PERCENTILE_CONT(0.5) WITHIN GROUP(ORDER BY life_expectancy) AS NUMERIC), 1) DESC
							    ) < 0.25 THEN '1st'
	   WHEN PERCENT_RANK() OVER(PARTITION BY region
								ORDER BY ROUND(CAST(PERCENTILE_CONT(0.5) WITHIN GROUP(ORDER BY life_expectancy) AS NUMERIC), 1) DESC
							    ) < 0.50 THEN '2nd'
	   WHEN PERCENT_RANK() OVER(PARTITION BY region
								ORDER BY ROUND(CAST(PERCENTILE_CONT(0.5) WITHIN GROUP(ORDER BY life_expectancy) AS NUMERIC), 1) DESC
							    ) < 0.75 THEN '3rd'
	   ELSE '4th'
	   END AS life_exp_quartiles
  FROM world_life_expectancy
 WHERE gdp != 0
    OR schooling != 0
 GROUP BY region, country
 ORDER BY region
),

median_gdp_schooling AS (
SELECT leb.region, leb.country, leb.median_life_exp, leb.life_exp_quartiles,
	   ROUND(CAST(PERCENTILE_CONT(0.5) WITHIN GROUP(ORDER BY wle.gdp) AS NUMERIC), 1) AS country_median_gdp,
	   ROUND(CAST(PERCENTILE_CONT(0.5) WITHIN GROUP(ORDER BY wle.schooling) AS NUMERIC), 1) AS country_median_schooling
  FROM life_exp_bins leb
  	   JOIN world_life_expectancy wle
		 ON leb.region = wle.region
		    AND leb.country = wle.country
 GROUP BY leb.region, leb.country, leb.median_life_exp, leb.life_exp_quartiles
 ORDER BY leb.region, leb.country, leb.median_life_exp, leb.life_exp_quartiles
)

SELECT region, life_exp_quartiles,
	   ROUND(CAST(PERCENTILE_CONT(0.5) WITHIN GROUP(ORDER BY country_median_gdp) AS NUMERIC), 1) AS region_median_gdp, 
	   ROUND(CAST(PERCENTILE_CONT(0.5) WITHIN GROUP(ORDER BY country_median_schooling) AS NUMERIC), 1) AS region_median_schooling
  FROM median_gdp_schooling
 GROUP BY region, life_exp_quartiles
 ORDER BY region, life_exp_quartiles
;

 * postgresql://postgres:***@localhost:5433/world_life_expectancy
20 rows affected.


[('Africa', '1st', Decimal('1176.0'), Decimal('11.8')),
 ('Africa', '2nd', Decimal('457.3'), Decimal('9.9')),
 ('Africa', '3rd', Decimal('422.0'), Decimal('8.8')),
 ('Africa', '4th', Decimal('430.0'), Decimal('8.6')),
 ('Americas', '1st', Decimal('5417.0'), Decimal('14.7')),
 ('Americas', '2nd', Decimal('1626.0'), Decimal('13.0')),
 ('Americas', '3rd', Decimal('1974.8'), Decimal('12.6')),
 ('Americas', '4th', Decimal('2262.0'), Decimal('12.1')),
 ('Asia', '1st', Decimal('15476.5'), Decimal('13.2')),
 ('Asia', '2nd', Decimal('2041.8'), Decimal('12.7')),
 ('Asia', '3rd', Decimal('1292.0'), Decimal('11.4')),
 ('Asia', '4th', Decimal('472.8'), Decimal('10.2')),
 ('Europe', '1st', Decimal('31783.0'), Decimal('16.1')),
 ('Europe', '2nd', Decimal('21774.5'), Decimal('16.3')),
 ('Europe', '3rd', Decimal('5836.5'), Decimal('14.7')),
 ('Europe', '4th', Decimal('4149.5'), Decimal('14.8')),
 ('Oceania', '1st', Decimal('9047.0'), Decimal('19.2')),
 ('Oceania', '2nd', Decimal('2101.3'), Decimal('12.6')),
 ('Oceania', '3rd', Decimal('1755.5'), Decimal('12.7')),
 ('Oceania', '4th', Decimal('801.5'), Decimal('9.1'))]

Across various regions, a nuanced correlation is evident among life expectancy, GDP, and schooling. Generally, a decline in GDP and schooling is associated with a decrease in life expectancy. However, it is essential to underscore that this relationship is only partially accurate, given that specific regions exhibit values that deviate from this general trend. Additionally, it's noteworthy that over 20% of all countries have been excluded from this comparison due to missing values in the relevant fields.


**NOTES**

- The first quartile within each region encompasses countries with the highest median life expectancy over time, whereas the fourth quartile includes those with the lowest median life expectancy over the same period.

- Countries with missing schooling and GDP values were excluded from this comparison, including:

Iran (Islamic Republic of), Viet Nam, Montenegro, Syrian Arab Republic, Papua New Guinea, United Kingdom of Great Britain and Northern Ireland, Sao Tome and Principe, Kyrgyzstan, Slovakia, Bosnia and Herzegovina, Lao People's Democratic Republic, Saint Lucia, Libya, Congo, Democratic Republic of the Congo, Republic of Korea, Democratic People's Republic of Korea, Venezuela (Bolivarian Republic of), Micronesia (Federated States of), Eritrea, Saint Vincent and the Grenadines, United Republic of Tanzania, Egypt, Bahamas, Timor-Leste, Gambia, Iraq, United States of America, Côte d'Ivoire, Yemen, South Sudan, Turkmenistan, The former Yugoslav republic of Macedonia, Bolivia (Plurinational State of), Republic of Moldova, Equatorial Guinea, Antigua and Barbuda, Czechia, Somalia.

(TOTAL: 39)

In [30]:
%%sql

-- Overal median GDP and schooling per mortality quartile by region
WITH
mortality_bins AS ( 
SELECT region, country, ROUND(CAST(PERCENTILE_CONT(0.5) WITHIN GROUP(ORDER BY adult_mortality) AS NUMERIC), 1) AS median_mortality,
	   CASE
	   WHEN PERCENT_RANK() OVER(PARTITION BY region
							    ORDER BY ROUND(CAST(PERCENTILE_CONT(0.5) WITHIN GROUP(ORDER BY adult_mortality) AS NUMERIC), 1) DESC
							    ) < 0.25 THEN '1st'
	   WHEN PERCENT_RANK() OVER(PARTITION BY region
								ORDER BY ROUND(CAST(PERCENTILE_CONT(0.5) WITHIN GROUP(ORDER BY adult_mortality) AS NUMERIC), 1) DESC
							    ) < 0.50 THEN '2nd'
	   WHEN PERCENT_RANK() OVER(PARTITION BY region
								ORDER BY ROUND(CAST(PERCENTILE_CONT(0.5) WITHIN GROUP(ORDER BY adult_mortality) AS NUMERIC), 1) DESC
							    ) < 0.75 THEN '3rd'
	   ELSE '4th'
	   END AS mortality_quartiles
  FROM world_life_expectancy
 WHERE gdp != 0
    OR schooling != 0
 GROUP BY region, country
 ORDER BY region
),

median_gdp_schooling AS (
SELECT mb.region, mb.country, mb.median_mortality, mb.mortality_quartiles,
	   ROUND(CAST(PERCENTILE_CONT(0.5) WITHIN GROUP(ORDER BY wle.gdp) AS NUMERIC), 1) AS country_median_gdp,
	   ROUND(CAST(PERCENTILE_CONT(0.5) WITHIN GROUP(ORDER BY wle.schooling) AS NUMERIC), 1) AS country_median_schooling
  FROM mortality_bins mb
  	   JOIN world_life_expectancy wle
		 ON mb.region = wle.region
		    AND mb.country = wle.country
 GROUP BY mb.region, mb.country, mb.median_mortality, mb.mortality_quartiles
 ORDER BY mb.region, mb.country, mb.median_mortality, mb.mortality_quartiles
)

SELECT region, mortality_quartiles,
	   ROUND(CAST(PERCENTILE_CONT(0.5) WITHIN GROUP(ORDER BY country_median_gdp) AS NUMERIC), 1) AS region_median_gdp, 
	   ROUND(CAST(PERCENTILE_CONT(0.5) WITHIN GROUP(ORDER BY country_median_schooling) AS NUMERIC), 1) AS region_median_schooling
  FROM median_gdp_schooling
 GROUP BY region, mortality_quartiles
 ORDER BY region, mortality_quartiles
;

 * postgresql://postgres:***@localhost:5433/world_life_expectancy
20 rows affected.


[('Africa', '1st', Decimal('888.5'), Decimal('9.8')),
 ('Africa', '2nd', Decimal('347.0'), Decimal('8.5')),
 ('Africa', '3rd', Decimal('559.0'), Decimal('9.0')),
 ('Africa', '4th', Decimal('1176.0'), Decimal('11.8')),
 ('Americas', '1st', Decimal('2262.0'), Decimal('12.5')),
 ('Americas', '2nd', Decimal('1498.5'), Decimal('13.0')),
 ('Americas', '3rd', Decimal('2517.8'), Decimal('12.9')),
 ('Americas', '4th', Decimal('5417.0'), Decimal('14.7')),
 ('Asia', '1st', Decimal('470.5'), Decimal('10.4')),
 ('Asia', '2nd', Decimal('607.0'), Decimal('10.8')),
 ('Asia', '3rd', Decimal('1826.5'), Decimal('12.0')),
 ('Asia', '4th', Decimal('17793.8'), Decimal('13.4')),
 ('Europe', '1st', Decimal('5836.5'), Decimal('15.0')),
 ('Europe', '2nd', Decimal('5445.5'), Decimal('14.7')),
 ('Europe', '3rd', Decimal('28942.5'), Decimal('16.3')),
 ('Europe', '4th', Decimal('29281.0'), Decimal('15.9')),
 ('Oceania', '1st', Decimal('1327.5'), Decimal('11.9')),
 ('Oceania', '2nd', Decimal('400.8'), Decimal('10.1')),
 ('Oceania', '3rd', Decimal('2101.3'), Decimal('12.6')),
 ('Oceania', '4th', Decimal('9047.0'), Decimal('19.2'))]

Across all regions, an inverse relationship seems to exist between GDP and schooling on one hand, and mortality on the other: as GDP and schooling increase, mortality tends to decrease. Nevertheless, it is essential to underscore that this relationship is only partially accurate, as certain regions exhibit values that deviate from this general trend. Additionally, it's noteworthy that over 20% of all countries have been excluded from this comparison due to missing values in the relevant fields.

**NOTES**

- The first quartile within each region encompasses countries with the highest median mortality over time, whereas the fourth quartile includes those with the lowest median mortality over the same period.

- Countries with missing schooling and GDP values were excluded from this comparison, including:

Iran (Islamic Republic of), Viet Nam, Montenegro, Syrian Arab Republic, Papua New Guinea, United Kingdom of Great Britain and Northern Ireland, Sao Tome and Principe, Kyrgyzstan, Slovakia, Bosnia and Herzegovina, Lao People's Democratic Republic, Saint Lucia, Libya, Congo, Democratic Republic of the Congo, Republic of Korea, Democratic People's Republic of Korea, Venezuela (Bolivarian Republic of), Micronesia (Federated States of), Eritrea, Saint Vincent and the Grenadines, United Republic of Tanzania, Egypt, Bahamas, Timor-Leste, Gambia, Iraq, United States of America, Côte d'Ivoire, Yemen, South Sudan, Turkmenistan, The former Yugoslav republic of Macedonia, Bolivia (Plurinational State of), Republic of Moldova, Equatorial Guinea, Antigua and Barbuda, Czechia, Somalia.

(TOTAL: 39)